pilot/pkg/model/service.go

```diff
@@ -1261,25 +1261,54 @@ func (s *Service) GetAddressForProxy(node *Proxy) string {
 // GetExtraAddressesForProxy returns a k8s service's extra addresses to the cluster where the node resides.
 // Especially for dual stack k8s service to get other IP family addresses.
 func (s *Service) GetExtraAddressesForProxy(node *Proxy) []string {
-	if features.EnableDualStack && node.Metadata != nil {
-		if node.Metadata.ClusterID != "" {
-			addresses := s.ClusterVIPs.GetAddressesFor(node.Metadata.ClusterID)
-			if len(addresses) > 1 {
-				return addresses[1:]
-			}
-		}
+	addresses := s.getAllAddressesForProxy(node)
+	if len(addresses) > 1 {
+		return addresses[1:]
 	}
 	return nil
 }
 
 // GetAllAddressesForProxy returns a k8s service's extra addresses to the cluster where the node resides.
 // Especially for dual stack k8s service to get other IP family addresses.
 func (s *Service) GetAllAddressesForProxy(node *Proxy) []string {
-	if (features.EnableDualStack || features.EnableAmbient) && node.Metadata != nil && node.Metadata.ClusterID != "" {
+	return s.getAllAddressesForProxy(node)
+}
+
+func (s *Service) getAllAddressesForProxy(node *Proxy) []string {
+	if node.Metadata != nil && node.Metadata.ClusterID != "" {
 		addresses := s.ClusterVIPs.GetAddressesFor(node.Metadata.ClusterID)
-		if len(addresses) > 0 {
+		if (features.EnableDualStack || features.EnableAmbient) && len(addresses) > 0 {
 			return addresses
 		}
+		var ipv4Addresses []string
+		var ipv6Addresses []string
+		for _, addr := range addresses {
+			if strings.Contains(addr, "/") {
+				if prefix, err := netip.ParsePrefix(addr); err != nil {
+					log.Warnf("failed to parse prefix address '%s': %s", addr, err)
+					continue
+				} else if prefix.Addr().Is4() {
+					ipv4Addresses = append(ipv4Addresses, addr)
+				} else if prefix.Addr().Is6() {
+					ipv6Addresses = append(ipv6Addresses, addr)
+				}
+			} else {
+				if ipAddr, err := netip.ParseAddr(addr); err != nil {
+					log.Warnf("failed to parse address '%s': %s", addr, err)
+					continue
+				} else if ipAddr.Is4() {
+					ipv4Addresses = append(ipv4Addresses, addr)
+				} else if ipAddr.Is6() {
+					ipv6Addresses = append(ipv6Addresses, addr)
```

Comment:
nit： for single stack, ignore the other check

Reply:
Done